ntp-proto/src/clock.rs

```diff
@@ -2,11 +2,11 @@ use std::time::SystemTime;
 
 use crate::NtpTimestamp;
 
-trait NtpClock {
+pub trait NtpClock {
     fn now(&self) -> NtpTimestamp;
 }
 
-struct SystemClock;
+pub struct SystemClock;
```

Comment:
I would at least start by spelling this as `SystemClock(())` so that this crate controls initialization.